packages/chart/src/hooks/useBarChart.ts

```diff
@@ -210,6 +230,20 @@ export const useBarChart = () => {
     if (config.legend.highlightOnHover && config.legend.behavior === 'highlight') setSeriesHighlight([])
   }
 
+  const updateBarsGroup = (barGroups, barWidth) => {
```

Comment:
pull this function out of the body of this wrapper function, export it and add test cases.